tests/x11/ImageMagick.pm

```diff
@@ -29,968 +29,972 @@ use testapi;
 use utils;
 
 sub run() {
-    select_console('x11');
-    x11_start_program("xterm");
+    select_console "x11";
+
+    x11_start_program "xterm";
+
     become_root;
-    type_string "while pgrep packagekitd; do pkcon quit; sleep 1; done \n";
-    zypper_call("in ImageMagick");
+    zypper_call "in ImageMagick";
     type_string "exit\n";
 
-    type_string "wget --quiet " . data_url('imagemagick/bg_script.sh') . " -O bg_script.sh \n";
+    assert_script_run "wget --quiet " . data_url('imagemagick/bg_script.sh') . " -O bg_script.sh";
     type_string "chmod +x bg_script.sh; ./bg_script.sh " . data_url('imagemagick/bg_script.sh') . " \n";
 
-    assert_screen("imagemagick_test", 10);
+    assert_screen "imagemagick_test";
     send_key "alt-f4";
     wait_idle;
```

Comment:
Yes, there are needles that look exactly the same in a row. Some images are the same although they are produced with different command. I could adjust it at some point to not have this behavior but not right now.